pandas/plotting/_core.py

```diff
@@ -365,6 +366,12 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError('no numeric data to plot')
 
+        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
```

Comment:
I think you can just call ``np.asarray`` for  all columns

Reply:
That doesn't work well. Tons of `SettingWithCopyError` would be raised when run `pandas/tests/plotting/test_frame.py`.

Reply:
use .assign

Reply:
`.assign` still doesn't work (if I don‘’t miss something). `.assign` requires `kwargs`, which is a dict of `{str. callable or Series}`, however in many dataframes the column label may not be a string (like `pd.DataFrame([1,2,3])`).

Reply:
Fixed now.